fecgov/openFEC
# Add filters to` /totals/entity_type/`  to support PAC/Party totals data table

## Summary (required)

- Resolves #4878

Add filters to /totals/entity_type/ to support PAC/Party totals data table

### Required reviewers

- @hcaofec for database changes
- @fec-jli and/or @pkfec for application changes

## Impacted areas of the application

General components of the application that this PR will affect:

### Database changes

- 1 - Modify `ofec_totals_combined_mv` and `ofec_totals_combined_vw` to add:
    `treasurer_text` (replaces V0228)

- 2 - Modify `ofec_totals_pac_party_vw` to add:
    `treasurer_text` (replaces V0228)

- 3 - Modify `ofec_committee_totals_per_cycle_vw` to add:
    `committee_state`
    `treasurer_name`,
    `treasurer_text`
    `filing_frequency`,
    `filing_frequency_full`,
    `first_file_date`
(replaces V0127)

- 4 - Modify `ofec_totals_house_senate_mv` to add:
    `committee_state`
    `treasurer_name`,
    `treasurer_text`
    `filing_frequency`,
    `filing_frequency_full`,
    `first_file_date`
(replaces V0211)

- 5 - Modify `ofec_totals_ie_only_mv` to add:
    `committee_state`
    `filing_frequency`,
    `filing_frequency_full`,
    `first_file_date`
(replaces V0107)

### Application changes
- Remove unused model `CommitteeTotalsPresidential` (MV is being used in other MV's so we need to keep that, but this model isn't being used by the application - replaced by `CommitteeTotalsPerCycle` a while back
- Add filters to /totals/entity_type/:
`committee_type`,
`committee_state`,
`treasurer_name`
`min_disbursements`,
`max_disbursements`,
`min_receipts`,
`max_receipts`,
`min_last_cash_on_hand_end_period`,
`max_last_cash_on_hand_end_period_amount`,
`min_last_debts_owed_by_committee`,
`max_last_debts_owed_by_committee`

## How to test
- Check out this branch
- Test migration file and MV refresh with `createdb cfdm_test` / `dropdb cfdm_test` / `invoke create_sample_db`
- Connect to the `dev` database
- Modify the following classes in your local `openFEC/webservices/common/models/totals.py` to point to the temporary testing views (creation script here: https://github.com/fecgov/openFEC/issues/4878#issuecomment-869799959)
```
 class CommitteeTotalsPacParty(CommitteeTotals):
-    __tablename__ = 'ofec_totals_pac_party_vw'
+    __tablename__ = 'ofec_totals_pac_party_vw_lb'

 class CommitteeTotalsHouseSenate(CommitteeTotals):
-    __tablename__ = 'ofec_totals_house_senate_mv'
+    __tablename__ = 'ofec_totals_house_senate_vw_lb'
 
class CommitteeTotalsIEOnly(BaseModel):
-    __tablename__ = 'ofec_totals_ie_only_mv'
+    __tablename__ = 'ofec_totals_ie_only_mv_lb'
 
 class CommitteeTotalsPerCycle(CommitteeTotals):
-    __tablename__ = 'ofec_committee_totals_per_cycle_vw'
+    __tablename__ = 'ofec_committee_totals_per_cycle_vw_lb'
```
- Run some test filter queries like these: https://docs.google.com/spreadsheets/d/1NIj_Fcmby10_wLIZ2wI4eOmVp47n08SN_keGZsWkDWY/edit#gid=0 (focus on pac-party entity)
- Test existing filters like `cycle` and `committee_id`

## Related PR
https://github.com/fecgov/openFEC/pull/3694 created `CommitteeTotalsPerCycle` model
